src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -304,1237 +234,653 @@ export default class Fms {
-     *
-     * @for Fms
-     * @method _findLegByRouteString
-     * @param routeString {string}
-     * @return {LegModel|undefined}
-     * @private
-     */
-    _findLegByRouteString(routeString) {
-        return _find(this.legCollection, { routeString: routeString.toLowerCase() });
+    * Fascade
+    *
+    * @for Fms
+    * @method skipToWaypointName
+    * @param waypointName {string}
+    */
+    skipToWaypointName(waypointName) {
+        return this._routeModel.skipToWaypointName(waypointName);
     }
 
-    /**
-     * This method will find an leg at `legIndex` in the `#legCollection` and
-     * replace it with a new `routeString`.
-     *
-     * It is important to note that this doesn't create a new `LegModel` instance.
-     * Instead, this locates the leg at `legIndex`, destroys it's properties, then
-     * runs `init()` with the new `routeString`.
-     *
-     * @for Fms
-     * @method _replaceLegAtIndexWithRouteString
-     * @param legIndex {number}     array index of the leg to replace
-     * @param routeString {string}  routeString to use for the replacement leg
-     * @private
-     */
-    _replaceLegAtIndexWithRouteString(legIndex, routeString) {
-        const legModel = this.legCollection[legIndex];
+    // ------------------------------ PRIVATE ------------------------------
 
-        legModel.destroy();
-        legModel.init(routeString, this.currentRunwayName, this.currentPhase);
-    }
+    // /**
+    // * Build a `LegModel` instance that contains a `WaypointModel` with hold properties
+    // *
+    // * @for Fms
+    // * @method _createLegWithHoldWaypoint
+    // * @param waypointProps {object}
+    // * @return legModel {LegModel}
+    // */
+    // _createLegWithHoldWaypoint(waypointProps) {
+    //     const legModel = new LegModel(
+    //         waypointProps.name,
+    //         this.currentRunwayName,
+    //         this.currentPhase,
+    //         this._navigationLibrary,
+    //         waypointProps
+    //     );
+    //
+    //     return legModel;
+    // }
 
     /**
-     * Regenerate SID leg based on current runway assignment
+     * Update the expected arrival runway based on the STAR's exit point runway
      *
      * @for Fms
-     * @method _regenerateSidLeg
+     * @method _updateArrivalRunwayFromRoute
      */
-    _regenerateSidLeg() {
-        const sidLegIndex = this._findLegIndexForProcedureType(PROCEDURE_TYPE.SID);
-
-        if (sidLegIndex === INVALID_INDEX) {
-            // would be throwing here, except that causes the initial setting of the
-            // departure runway to throw. So instead will just return early.
-            return;
-        }
-
-        const sidLeg = this.legCollection[sidLegIndex];
-
-        this.replaceDepartureProcedure(sidLeg.routeString, this.departureRunwayModel);
-    }
+    _updateArrivalRunwayFromRoute() {
+        const arrivalRunwayModel = this._routeModel.getArrivalRunwayModel();
```

Comment:
without having seen the `RouteModel`, I feel like this should be a simple getter.